src/robot/client.ts

```diff
@@ -336,6 +341,37 @@ export class RobotClient implements Robot {
          * read and then write to 'peerConn', even after we have awaited/paused.
          */
         this.peerConn = webRTCConn.peerConnection; // eslint-disable-line require-atomic-updates
+        this.peerConn.addEventListener('iceconnectionstatechange', () => {
+          /*
+           * TODO: are there any disconnection scenarios where we can reuse the
+           * same connection and restart ice?
+           *
+           * All connection loss scenarios I tested seem to result in the peer
+           * connection getting closed, so restarting ice is not a valid way to
+           * recover.
```

Comment:
@edaniels maybe you have some insight here - it seems like all connection loss scenarios will result in a peer connection close. I'm guessing this semantic is required for webrtc over grpc-web? 

Ideally I would like to reuse the existing peer connection, but obviously this is not possible once it's closed.

Reply:
I think our usage of ICE restart is very low and we almost always end up reconnecting. I think for now, we should reset our peer connection and make a new one when we've detected a disconnection. It's likely the server side need works to support ICE restart in addition to utilizing the established negotiation channel for SDP retransmission.